test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1

```diff
@@ -0,0 +1,653 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "Extended FileSystem Provider Tests for Get-ChildItem cmdlet" -Tags "CI" {
+    BeforeAll {
+        $restoreLocation = Get-Location
+
+        $DirSep = [IO.Path]::DirectorySeparatorChar
+
+        $rootDir = Join-Path "TestDrive:" "TestDir"
+        New-Item -Path $rootDir -ItemType Directory > $null
+
+        Set-Location $rootDir
+
+        New-Item -Path "file1.txt" -ItemType File > $null
```

Comment:
Would it be better to have the filenames in a hashtable with their attribute so that you can verify the filenames are returned later in the tests?  Then you could also just loop through the hashtable to create the files instead of having separate lines.

Reply:
I see your point but I don't want complicate the test code because (1) there are issues we want fix in near future, (2) there are issues we want fix in FileSystem provider V2 future - so simple linear tests will simplify our work in future.